torch/distributed/nn/api/remote_module.py

```diff
@@ -0,0 +1,245 @@
+        )
+    else:
+        is_scriptable = False
+        remote_module_cls = _NON_SCRIPTABLE_REMOTE_MODULE_CLS
+
+        # Create the module on the remote side.
+        module_rref = rpc.remote(to, module_creator, args, kwargs)
+
+    # Create remote_module_cls instance on local side.
+    remote_module = remote_module_cls(module_rref, is_scriptable, global_unique_name)
+
+    return remote_module
+
+
+def RemoteModule(
+    to: str,
+    module_creator: Callable,
+    args: Tuple = None,
+    kwargs: Dict[str, Any] = None,
+    global_unique_name: str = None,
+):
+    """
+        A RemoteModule instance can only be created after RPC initialization.
+        It creates a user-specified module on a specified remote node.
+        It behaves like a regular nn.Module except that the ``forward`` method is
+        executed on the remote node.
+        It takes care of autograd recording to ensure the backward pass propogates
+        gradients back to the corresponding remote module.
+
+        The arguments of ``forward_async`` and ``forward`` are the same as
+        the ``forward`` method of the module returned by the ``module_creator``.
+
+        For example, if ``module_creator`` returns an instace of ``nn.Linear``,
+        that has ``forward`` method signature, ``def forward(input: Tensor) -> Tensor:``,
+        the generated `RemoteModule`` will have 2 methods in signature of
+        ``def forward(input: Tensor) -> Tensor:`` and
+        ``def forward_async(input: Tensor) -> Future[Tensor]:``.
+
+    Arguments:
+        to (str or WorkerInfo): id or name of the destination worker.
+        module_creator (Callable): A ``module_creator`` could be
+                1. A type object that is subclass of ``nn.Module``.
+                    For example,
+                    >>> class MyModule(nn.Module):
+                    >>>     def forward(input):
+                    >>>         return input + 1
+                    >>>
+                    >>> module_creator = MyModule
+                2. A function that returns a instance of ``nn.Module``.
+                    For example,
+                    >>> def module_creator():
+                    >>>     module = MyModule()
+                    >>>     scripted_module = torch.jit.script(module)
+                    >>>     return scripted_module
+        args (Sequence, optional): args to be passed to ``module_creator``.
+        kwargs (Dict, optional): kwargs to be passed to ``module_creator``.
+        global_unique_name (str, optional): The unique name of the created RemoteModule,
+            useful for profiling purpose. If not provided, a UUID4 will
+            be generated as its name.
+
+    Returns:
+        A remote module instance which wraps the :class:`~nn.Module` created by the
+        user-provided ``module_creator``, it has a blocking ``forward`` method and an asynchronous
+        ``forward`` that returns a future of the ``forward`` call on the user created
```

Comment:
the first one should be ``forward_async``?

Reply:
Updated.